### ISSUE TITLE
Terragrunt modules that include `terraform {source = }` from a common module are not included in atlantis config

### ISSUE THREAD

**Issue by fvdnabee** [transcend-io/terragrunt-atlantis-config#196]
When generating an atlantis config for the [latest gruntwork infrastructure live example repo](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/tree/6942725006e5f02fb1aedc3d3ea22aafb040d7d9), the output does not include projects for TG modules without a `terraform { source }` block. In terragrunt-infrastructure-live-example, there are some TG modules that reference [_envcommon](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/tree/6942725006e5f02fb1aedc3d3ea22aafb040d7d9/_envcommon) for sharing components across environments. These modules are missing from TCA's output.

Note the current [test_examples/terragrunt-infrastructure-live-example](https://github.com/transcend-io/terragrunt-atlantis-config/tree/master/test_examples/terragrunt-infrastructure-live-example) does not use this _envcommon structure, so this issue does not present itself in that test example. However, given that gruntwork has published a new example with this _envcommon structure, I'm guessing it will be picked up more and more.

Output on my system:
```
❯ git clone https://github.com/gruntwork-io/terragrunt-infrastructure-live-example
❯ terragrunt-atlantis-config generate
INFO[0000] Could not find an old config file. Starting from scratch 
INFO[0000] Created project for /home/fvdnabee/terragrunt-infrastructure-live-example/non-prod/us-east-1/qa/webserver-cluster/terragrunt.hcl 
INFO[0000] Created project for /home/fvdnabee/terragrunt-infrastructure-live-example/non-prod/us-east-1/qa/mysql/terragrunt.hcl 
INFO[0000] automerge: false
parallel_apply: true
parallel_plan: true
projects:
- autoplan:
    enabled: false
    when_modified:
    - '*.hcl'
    - '*.tf*'
  dir: non-prod/us-east-1/qa/mysql
- autoplan:
    enabled: false
    when_modified:
    - '*.hcl'
    - '*.tf*'
  dir: non-prod/us-east-1/qa/webserver-cluster
version: 3 
```
These two projects are included in the config because the `qa/webserver-cluster` and `qa/mysql` TG modules include a `terraform { source = }` block (these modules override the common configuration). All other TG modules are expected to be present as projects, as they include a module from _envcommon that contains a terraform source block, but are missing from the output. 

Are there some CLI options that instruct TAC to output all projects? Is there a temporary workaround that could help here? I'm using a very similar project layout with an _envcommon folder and am keen on getting TAC working.

terragrunt-infrastructure-live-example tree:
```
❯ tree
.
├── CODEOWNERS
├── _envcommon
│   ├── mysql.hcl
│   ├── README.md
│   └── webserver-cluster.hcl
├── LICENSE.txt
├── non-prod
│   ├── account.hcl
│   └── us-east-1
│       ├── qa
│       │   ├── env.hcl
│       │   ├── mysql
│       │   │   └── terragrunt.hcl
│       │   └── webserver-cluster
│       │       └── terragrunt.hcl
│       ├── region.hcl
│       └── stage
│           ├── env.hcl
│           ├── mysql
│           │   └── terragrunt.hcl
│           └── webserver-cluster
│               └── terragrunt.hcl
├── prod
│   ├── account.hcl
│   └── us-east-1
│       ├── prod
│       │   ├── env.hcl
│       │   ├── mysql
│       │   │   └── terragrunt.hcl
│       │   └── webserver-cluster
│       │       └── terragrunt.hcl
│       └── region.hcl
├── README.md
└── terragrunt.hcl
```

Version:
```
❯ terragrunt-atlantis-config version
terragrunt-atlantis-config '1.12.3'
```

Seemingly related issue: #120

**Comment by fvdnabee** [comment]
After some digging, I've found the following:
* `terragrunt-atlantis-config` does not handle named include blocks correctly. It appears to ignore their presence altogether. With some patching to `cmd/parse_hcl.go`, you can improve this handling. I think this is something worth pursuing as the [Terragrunt docs](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#include) state `You can have more than one include block, but each one must have a unique label. It is recommended to always label your include blocks. Bare includes (include block with no label - e.g., include {}) are currently supported for backward compatibility, but is deprecated usage and support may be removed in the future.` I have a simple patch, that seems to work ok (didn't test all the examples however).
* Given this shortcoming, you can work around it by having at least one include block without a label in your TG module. 

Additionally, for the vanilla terragrunt-infrastructure-live-example TG repo you will also want to add `extra_atlantis_dependencies` statements to signal the dependency on the _envcommon modules.

Taking `non-prod/us-east-1/stage/mysql/terragrunt.hcl` as an example: after altering the module as follows:
```diff
--- a/non-prod/us-east-1/stage/mysql/terragrunt.hcl
+++ b/non-prod/us-east-1/stage/mysql/terragrunt.hcl
@@ -10,7 +10,7 @@
 
 # Include the root `terragrunt.hcl` configuration. The root configuration contains settings that are common across all
 # components and environments, such as how to configure remote state.
-include "root" {
+include {
   path = find_in_parent_folders()
 }
 
@@ -24,3 +24,8 @@ include "envcommon" {
 # ---------------------------------------------------------------------------------------------------------------------
 # We don't need to override any of the common parameters for this environment, so we don't specify any other parameters.
 # ---------------------------------------------------------------------------------------------------------------------
+locals {
+  extra_atlantis_dependencies = [
+    "${dirname(find_in_parent_folders())}//_envcommon/mysql.hcl"
+  ]
+}
```
The following project is added to the generated config (which appears to be correct):
```
- autoplan:
    enabled: false
    when_modified:
    - '*.hcl'
    - '*.tf*'
    - ../../../account.hcl
    - ../../region.hcl
    - ../env.hcl
    - ../../../../_envcommon/mysql.hcl
  dir: non-prod/us-east-1/stage/mysql
```

**Comment by dmattia** [comment]
This is great. I'd love to see the patch you have so far. In general, these sorts of things I hope are fixed just by bumping up Terragrunt's version in this lib, which tries to lean on the external tool's parsing as much as possible, but it's not always possible.

Also, feel free to add a PR with the infra-live example with the changes you describe in the test examples folder, and I can help update the code to make that test case pass (along with all the others).

We actually have an older version of the live repo in the test examples right now, but that was quite a ways before multiple includes and named includes were added. So this is definitely something I think TAC should support

**Comment by barrydobson** [comment]
I've opened https://github.com/transcend-io/terragrunt-atlantis-config/pull/201 to address this. It has a failing test though. Help appreciated

**Comment by dmattia** [comment]
This seems to have been closed by #202
